### PR TITLE
Add driver for eCO2 and eTVOC sensor CCS811

### PR DESCRIPTION
**Pull request by RafaelLeeImg**

### CODE REVIEW

File: pslab/external/ccs811.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        raw_voltage = (raw_data & ((1 << 10) - 1)) * (1.65 / 1023)
+
+        result = {"eCO2": eCO2, 'eTVOC': eTVOC, 'status': status, 'error_id': error_id}
+
+        if error_id == 0:
+            pass
+        elif error_id == 1:
+            print("The CCS811 received an I²C read request to a mailbox ID that is invalid")
+        else:
+            print("Other error")
```
Comment (bessman) [inline pslab/external/ccs811.py:138]: These should `raise RuntimeError`. The user can decide whether to catch the exception.

File: pslab/external/ccs811.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        return result
+
+
+def connect(route, **args):
+    return TSL2561(route, **args)
```
Comment (bessman) [inline pslab/external/ccs811.py:144]: ?

File: pslab/external/ccs811.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        print(f'hex(hardware_id) = {hex(hardware_id)}')
+        time.sleep(0.02)  # 20ms
+        hardware_version = (self.read(1, self._HW_Version))[0]
+        print(f'hex(hardware_version) = {hex(hardware_version)}')
+        time.sleep(0.02)  # 20ms
+        boot_version = (self.read(2, self._FW_Boot_Version))[0]
+        print(f'hex(boot_version) = {hex(boot_version)}')
+        time.sleep(0.02)  # 20ms
+        app_version = (self.read(2, self._FW_App_Version))[0]
+        print(f'hex(app_version) = {hex(app_version)}')
```
Comment (bessman) [inline pslab/external/ccs811.py:84]: These values should be returned, not printed.

**Review by bessman — CHANGES_REQUESTED** [review]

**Comment by RafaelLeeImg** [comment]
pipeline failed:
tox -e lint

lint: tox-gh-actions couldn't understand the parallel option. ignoring the given option: 0
tox: lint
  lint: commands[0]> black --check pslab/bus/ pslab/instrument/ pslab/serial_handler.py pslab/cli.py pslab/external/motor.py pslab/external/gas_sensor.py pslab/external/hcsr04.py
  would reformat /home/runner/work/pslab-python/pslab-python/pslab/bus/spi.py

File: pslab/external/ccs811.py (hunk 2 above)
Comment (RafaelLeeImg) [reply pslab/external/ccs811.py]: Deleted in 6b00a77.

File: pslab/external/ccs811.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    # 2:1 -
+    _REG_STATUS_BIT_ERROR = 0
+
+    def __init__(self, **args):
+        self._ADDRESS = args.get('address', self._ADDRESS)
+        super().__init__(self._ADDRESS)
+        self.fetchID()
+        self.softwareReset()
+
+    def softwareReset(self):
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply pslab/external/ccs811.py:53]: A large portion of code are not using snake_case. Are you really sure about that?

    def initTemperature(self):
    def readRegister(self, register):
    def writeRegister(self, reg, value):
    def setGain(self, gain):
    def setChannel(self, channel):
    def setDataRate(self, rate):
    def getRaw(self):
    def getAccel(self):
    def getTemp(self):
    def getGyro(self):
    def getMag(self):

File: pslab/external/ccs811.py (hunk 3 above)
Comment (RafaelLeeImg) [reply pslab/external/ccs811.py]: Fixed in 6b00a77

File: pslab/external/ccs811.py (hunk 1 above)
Comment (RafaelLeeImg) [reply pslab/external/ccs811.py]: Fixed in 6b00a77.

File: pslab/external/ccs811.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    # Bit(s) Field
+    _REG_STATUS_BIT_FW_MODE = 7
+    _REG_STATUS_BIT_APP_ERASE = 6
+    _REG_STATUS_BIT_APP_VERIFY = 5
+    _REG_STATUS_BIT_APP_VALID = 4
+    _REG_STATUS_BIT_DATA_READY = 3
+    # 2:1 -
+    _REG_STATUS_BIT_ERROR = 0
+
+    def __init__(self, **args):
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply pslab/external/ccs811.py]: Fixed in 84fdf83.

File: pslab/external/ccs811.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    _MEAS_MODE = 0x01
+    _ALG_RESULT_DATA = 0x02  # ALG_RESULT_DATA # R 8 bytes the next two bytes contain a ppb estimate of the total VOC level. Raw ADC data values for resistance and current source
+    _RAW_DATA = 0x03  # RAW_DATA # R 2 bytes used. Temperature and humidity data can be written to
+    _ENV_DATA = 0x05  # ENV_DATA # W 4 bytes enable compensation Thresholds for operation when interrupts are only
+    _THRESHOLDS = 0x10  # THRESHOLDS # W 4 bytes generated when eCO2 ppm crosses a threshold The encoded current baseline value can be read. A
+    # BASELINE # R/W 2 bytes previously saved encoded baseline can be written.
+    _BASELINE = 0x11
+    _HW_ID = 0x20  # HW_ID # R 1 byte Hardware ID. The value is 0x81
+    _HW = 0x21  # HW Version # R 1 byte Hardware Version. The value is 0x1X Firmware Boot Version. The first 2 bytes contain the
+    _FW_Boot_Version = 0x23  # FW_Boot_Version # R 2 bytes firmware version number for the boot code. Firmware Application Version. The first 2 bytes contain
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply pslab/external/ccs811.py]: My mistake. Fixed in 6b00a77

**Comment by bessman** [comment]
> pipeline failed: tox -e lint

Fixed, please rebase.

File: pslab/external/ccs811.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (bessman) [reply pslab/external/ccs811.py]: See https://github.com/fossasia/pslab-python/commit/84fdf8324bf8b3cb60341f977d7edcbea0faaffb#r114593454

File: pslab/external/ccs811.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply pslab/external/ccs811.py:52]: Coding style is fixed in 358bce3.

**Comment by RafaelLeeImg** [comment]
Is there anything more I need to fix?

**Pull request merged**